client/asset/dcr/dcr.go

```diff
@@ -1169,12 +1257,24 @@ func (dcr *ExchangeWallet) fund(enough func(sum uint64, size uint32, unspent *co
 
 // spendableUTXOs generates a slice of spendable *compositeUTXO.
 func (dcr *ExchangeWallet) spendableUTXOs() ([]*compositeUTXO, error) {
-	unspents, err := dcr.wallet.Unspents(dcr.ctx, dcr.acctName)
+	unspents, err := dcr.wallet.Unspents(dcr.ctx, dcr.primaryAcct)
 	if err != nil {
 		return nil, err
 	}
+	if dcr.tradingAccount != "" {
+		// Trading account may contain spendable utxos such as
+		// unspent split tx outputs that are unlocked/returned.
+		// TODO: Care should probably be taken to ensure only
+		// unspent split tx outputs are selected and other
+		// unmixed outputs in the trading account are ignored.
+		tradingAcctSpendables, err := dcr.wallet.Unspents(dcr.ctx, dcr.tradingAccount)
+		if err != nil {
+			return nil, err
+		}
+		unspents = append(unspents, tradingAcctSpendables...)
```

Comment:
Should these include only the external branch utxos from the temp trading account?

Or is it assumed that any coins in the internal branch would be locked, and when unlocked they would be immediately sent to the unmixed account?

Is this what the TODO comment is getting at?

Reply:
As an aside, I feel like we should explicitly document on the `Wallet interface` that the `Unspents` method must not return any locked outputs.  We're relying on that right?

Reply:
Yes, we are relying on that. Should doc, I suppose. Will also need to confirm that's the current behaviour, although I believe it is.

> is it assumed that any coins in the internal branch would be locked, and when unlocked they would be immediately sent to the unmixed account?

Yes, but looking at it again, probably not a good assumption since the transfer to the unmixed account could fail (in theory at least) and if `wallet.Unspent` is called at some point between unlocking the coins and making the transfer to the unmixed account.

> Should these include only the external branch utxos from the temp trading account?

I thought about it, maybe. But is it an unnecessary overhead? Because we'd need a validateaddress rpc for each unspent output. Given the previous point about potentially selecting tainted utxos meant for the unmixed account, should probably be worth it to make this check.

There's also the concern about the user sending funds directly to this temp account and ends up being selected for funding orders because the fund was sent to an external branch address and is not locked. Will think up a potentially better way of uniquely identifying unused split tx outputs.

Reply:
> There's also the concern about the user sending funds directly to this temp account and ends up being selected for funding orders because the fund was sent to an external branch address and is not locked. Will think up a potentially better way of uniquely identifying unused split tx outputs.

I'm not sure we should jump through too many hoops for this case.  It's a similar situation with dcrwallet and the user receiving funds directly into the mixed account instead of the unmixed account.  Just user error.  But maybe...

Reply:
I suppose also if `sendAll` did have an error, the user would have to manually rescue those funds if we did not permit unlocked internal branch utxos to be selected